src/borda_clj/client.clj

```diff
@@ -0,0 +1,89 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(def submit-key {:op "_submit"})
+
+(defn submit-failed [measurements values]
+  (update measurements submit-key (partial merge-with +) {:error_count (get values :_submits)}))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and
+   caps the size of the hashmap at the given max-buffer-size."
+  [measurements max-buffer-size dimensions values]
+  (let [vals (merge {:_submits 1} values)]
+    (if (contains? measurements dimensions)
+      ; TODO: support stuff other than SUM here (e.g. AVG, MIN, MAX, etc.)
+      (update measurements dimensions (partial merge-with +) vals) ; merge with existing in buffer
+      (if (< (count measurements) max-buffer-size)
+        (assoc measurements dimensions vals) ; space available, add to buffer
+        (do (println "borda buffer full, discarding measurement" dimensions vals)
+            (submit-failed measurements vals)))))) ; buffer full
+
+(defn remove-submits [[dimensions values]]
+  [dimensions (dissoc values :_submits)])
+
+(defn finalize-submit-counts [measurements]
+  (let [m (update measurements submit-key merge {:success_count (reduce + (map (fn [val] (get val :_submits)) (vals measurements)))})]
```

Comment:
Just `:_submits` is a shorter way to say `(fn [val] (get val :_submits))`.  In clojure, keywords can be used wherever you'd use a function name or `fn` expression, with exactly these semantics.